internal/controllers/machineset/machineset_controller.go

```diff
@@ -290,6 +295,7 @@ func (r *Reconciler) reconcile(ctx context.Context, cluster *clusterv1.Cluster,
 		filteredMachines = append(filteredMachines, machine)
 	}
 
+	// TODO: This logic can also be moved into computeDesiredMachine function.
```

Comment:
Do we end up with co-ownership if we don't move it?

Reply:
Should we do the same in the MD controller?

Reply:
I lean towards moving all writes that we do in the MD controller (on MachineSets) and in the MS controller (on Machines) into computeDesired.

Except if there is a reason why we cannot do it.

Reply:
[WIP]

Taking a closer look:

MD controller:
* additionall we patch the MachineDeploymentNameLabel, but we only do it if it's not already there
  * It's not enough to move this into computeDesiredMachineSet as we don't update all MachineSets just "new"

Reply:
Looks like we have stuff like `scaleMachineSet` where we try to write some fields with the regular patchHelper. I wonder if that works